bambou/session.go

```diff
@@ -121,14 +130,25 @@ func (s *Session) makeAuthorizationHeaders() (string, *Error) {
 
 func (s *Session) prepareHeaders(request *http.Request, info *FetchingInfo) *Error {
 
-	authString, berr := s.makeAuthorizationHeaders()
-	if berr != nil {
-		return berr
+	if s.Certificate != nil { // We're using X509 certificate based auth.
+
+		// XXX - "InsecureSkipVerify"
+		s.client.Transport = &http.Transport{TLSClientConfig: &tls.Config{Certificates: []tls.Certificate{*s.Certificate}, InsecureSkipVerify: true}}
+
+	} else { // We're using user & password based authentication
+
+		// Skip TLS certificate verification
+		s.client.Transport = &http.Transport{TLSClientConfig: &tls.Config{InsecureSkipVerify: true}}
+		authString, err := s.makeAuthorizationHeaders()
+		if err != nil {
+			return err
+		}
+		request.Header.Set("Authorization", authString)
+		request.Header.Set("X-Nuage-Organization", s.Organization)
 	}
 
-	request.Header.Set("Authorization", authString)
+	// Common headers
 	request.Header.Set("X-Nuage-PageSize", "50")
-	request.Header.Set("X-Nuage-Organization", s.Organization)
```

Comment:
This header is not required for certificate based auth? (not sure, so asking :))